pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -220,11 +220,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			return r.reconcile(ctx, log, addon, cluster)
 		},
 	)
+
 	if err != nil {
 		r.recorder.Event(addon, corev1.EventTypeWarning, "ReconcilingError", err.Error())
 		r.recorder.Eventf(cluster, corev1.EventTypeWarning, "ReconcilingError",
 			"failed to reconcile Addon %q: %v", addon.Name, err)
-	} else if result == nil {
+
```

Comment:
nit: would be great if you could remove this newline from here but I can live with that as well.